### DISCUSSION TITLE
Regarding helics_cli terminal

### DISCUSSION THREAD

**Discussion by pratyush0101** [GMLC-TDC/HELICS#2224] [Q&A] [answered]
Hello,

I was trying out the base examples and did not understand the part about installing helics_cli. The reason may be that I am using python for the very first time. I have been trying the base example using the jupyter notebook. I installed helics_cli using pip install [pip install git+git://github.com/GMLC-TDC/helics-cli.git@main](url)
I don't know how to access the helics_cli terminal. How can I access and use the helics_cli terminal from the jupyter notebook?

**Comment by trevorhardy** [comment, marked as answer]
Oh, this is a good question. I'm not a Jupyter notebook user but I suspect there is a way that you can run a system or shell command. Let me look really quick...


[From here](https://coderzcolumn.com/tutorials/python/list-of-useful-magic-commands-in-jupyter-notebook-lab), it looks like Jupyter has a magic system command to let you do things like this.

%system
The %system command lets us execute Unix shell commands in the jupyter notebook. We can execute any single line Unix shell command from the notebook. We have explained below the usage of the command with two simple examples.

```
%system echo 'Hello World'
['Hello World']
```

So you should be able to do something like:

```
%system helics run --path=./<helics_cli runner JSON>
```

**Reply by pratyush0101** [reply]
I wrote the following code
```
import helics
import helics_cli
%system helics run --path=fundamental_default_runner.json
```
but got the following traceback error:
```'Traceback (most recent call last):',

 '  File "e:\\anaconda\\lib\\runpy.py", line 194, in _run_module_as_main',
 '    return _run_code(code, main_globals, None,',

 '  File "e:\\anaconda\\lib\\runpy.py", line 87, in _run_code',
 '    exec(code, run_globals)',
 '  File "E:\\anaconda\\Scripts\\helics.exe\\__main__.py", line 4, in <module>',
 '  File "e:\\anaconda\\lib\\site-packages\\helics_cli\\cli.py", line 108, in <module>',
 '    def profile_plot(path, save, invert):',
 '  File "e:\\anaconda\\lib\\site-packages\\click\\decorators.py", line 192, in decorator',
 '    _param_memo(f, OptionClass(param_decls, **option_attrs))',
 '  File "e:\\anaconda\\lib\\site-packages\\click\\core.py", line 1714, in __init__',
 '    Parameter.__init__(self, param_decls, type=type, **attrs)',
 "TypeError: __init__() got an unexpected keyword argument 'prompt_required'"
```

What should I do to fix this? I tried installing a new version of `click` but that didn't resolve the error.

**Reply by kdheepak** [reply]
Can you check that you have `click` version 8 installed?

**Reply by pratyush0101** [reply]
Yes, it is the 8.0.1 version.

**Reply by kdheepak** [reply]
I think you may have multiple versions installed somehow. See this line for example: https://github.com/pallets/click/blob/8.0.1/src/click/core.py#L1714

In v8.0.1, line 1714 is different from line 1714 in your traceback:

```
 '  File "e:\\anaconda\\lib\\site-packages\\click\\core.py", line 1714, in __init__',
 '    Parameter.__init__(self, param_decls, type=type, **attrs)',
 ```

I would say instead of using the `%system` ipython magic command, use the following instead:

```python
import subprocess
import shlex

cmd = "helics run --path=fundamental_default_runner.json"

subprocess.call(shlex.split(cmd))
```

It's a few more lines, but should work better.

**Reply by pratyush0101** [reply]
When I run the following code

```
import helics
import helics_cli 
import subprocess
import shlex

cmd = "helics run --path=fundamental_default_runner.json"

subprocess.call(shlex.split(cmd))

```

I don't get the previous traceback error message. After executing this block of code, the output is displayed as `1`. According to the Base Example Co-simulation user guide [https://docs.helics.org/en/latest/user-guide/examples/fundamental_examples/fundamental_default.html](url), two graphs should be displayed after exceuting the `helics run --path=fundamental_default_runner.json` statement. 

Why am I not getting these two graphs?

**Reply by kdheepak** [reply]
It probably didn't run then, can you check the log files? Without the stdout output of the `subprocess.call` function and without the output of the log files, it's hard for me to tell what is going on.

I should say that `helics-cli` was really meant to be used from the command line. Trying this out from a notebook is going to result in behaviour that is related to how Python works and not necessarily how HELICS works.

**Reply by pratyush0101** [reply]
I don't know if I can ask this question here. I used `helics-cli` from the command line and there were no errors. Running the .json file showed the two outputs properly. The main reason why I am trying to use HELICS is to co-simulate between python and GridLAB-D. I have been trying to install GridLAB-D with HELICS for the past few days but have not been able to do so. Here are the steps that I followed:
1. I installed MSYS2.
2. added the mingw64/bin to path
3. installed all the necessary packages as specified on the windows-installation guide.
4. `git clone https://github.com/GMLC-TDC/HELICS.git helics_source`
5. `cd helics_source`
6. `git checkout tags/v3.1.1`
7. `mkdir helics-build`
8. `cd helics-build`
9.  `cmake-gui ../` (I checked the enable shared libraries option during configuration)
10.  `make -j`
11.  `make install`
12.  `git clone https://github.com/gridlab-d/gridlab-d.git GLD_clone`
13.  `git checkout release/v4.3`
14.  `autoreconf -if`
15.  `export PATH=$PATH:/e/HELICS`
16.   `./configure --build=x86_64-w64-mingw32 --prefix=/e/GridLAB-D_new --with-helics=/e/HELICS --with-xerces=/e/msys2/mingw64 --enable-silent-rules 'CFLAGS=-g -O2 -w' 'CXXFLAGS=-g -O2 -w' 'LDFLAGS=-g -O2 -w -L/mingw64/bin'`

During the configuration `checking for HELICS... no` message is displayed and I haven't been able to install GridLAB-D with HELICS. What can be the problem here?

**Reply by trevorhardy** [reply]
GridLAB-D release 4.3 doesn't support HELICS v3. Try using branch "develop".

**Comment by trevorhardy** [comment]
Its been two months so I'm going to close this one out for now.
